source/common/network/io_socket_handle_impl.cc

```diff
@@ -16,6 +18,48 @@ using Envoy::Api::SysCallSizeResult;
 namespace Envoy {
 namespace Network {
 
+namespace {
+
+void getSocketAddressInfo(const Address::Instance& address, sockaddr_storage& addr, socklen_t& sz) {
```

Comment:
this method is something I was trying to avoid in my earlier iterations.
I am trying to get the the sockaddr from the Address::Instance - but given the sockaddr lives inside the AddressImpl implementation, ideally, we should've been able to use it somehow.

Any other  recommendations on how to get this?

Reply:
Sorry, I should have got to this earlier.
Currently this file and address_impl.h both uses POSIX API. I think down_cast Address::Instance to Ipv(4|6)Instance is acceptable in this file. And adding Ipv(4|6)Instance::getSockaddr() is okay too. I assume for Windows, we will add another set of implementations of IoHandle and Address::Instance.
I would like to hear opinions from @jmarantz on this.

Reply:
I am pretty sure we cannot do that because it will cause a circular reference.

address_lib already depends on io_socket_handle_lib as it creates an instance of iosocket on the socket() call.

Reply:
circular dependency can be fixed by adjusting build targets though. What I want to confirm with @jmarantz is whether this approach is compatible with adding Windows support.

Reply:
I think we can ignore Windows for now.